src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_VALUE = 2;
+    private Node<K, V>[] table;
+    private int capacity;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        this.capacity = DEFAULT_CAPACITY;
+        this.threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+            return;
+        }
+        while (node.next != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        if (Objects.equals(node.key, key)) {
+            node.value = value;
+            return;
+        }
+        node.next = new Node<>(key, value, null);
+        size++;
 
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % capacity;
+    }
+
+    private void resize() {
+        capacity = capacity * RESIZE_VALUE;
+        Node<K, V>[] oldTable = table;
+        table = new Node[capacity];
+        size = 0;
+        for (Node<K, V> node : oldTable) {
+            Node<K, V> currentNode = node;
+            while (currentNode != null) {
+                put(currentNode.key, currentNode.value);
+                currentNode = currentNode.next;
+            }
+        }
+        threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+    }
+
+    private class Node<K, V> {
+        private final K key;
```

Comment:
```suggestion
        private K key;
```
i don't think that need to make it final